lib/openhab/core/types/time_series.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+# @deprecated OH4.0 this guard is not needed on OH4.1
+if Gem::Version.new(OpenHAB::Core::VERSION) >= Gem::Version.new("4.1.0")
+
+  require "forwardable"
+
+  module OpenHAB
+    module Core
+      module Types
+        TimeSeries = org.openhab.core.types.TimeSeries
+
+        #
+        # {TimeSeries} is used to transport a set of states together with their timestamp.
+        #
+        # The states are sorted chronologically. The entries can be accessed like an array.
+        #
+        # @since openHAB 4.1
+        #
+        # @example
+        #   time_series = TimeSeries.new(:add)
+        #                           .add(Time.at(2), DecimalType.new(2))
+        #                           .add(Time.at(1), DecimalType.new(1))
+        #                           .add(Time.at(3), DecimalType.new(3))
+        #   logger.info "first entry: #{time_series.first.state}" # => 1
+        #   logger.info "last entry: #{time_series.last.state}" # => 3
+        #   logger.info "second entry: #{time_series[1].state}" # => 2
+        #   logger.info "sum: #{time_series.sum(&:state)}" # => 6
+        #
+        # @see DSL::Rules::BuilderDSL#time_series_updated
+        #
+        class TimeSeries
+          extend Forwardable
+
+          # @!attribute [r] policy
+          #   Returns the persistence policy of this series.
+          #   @see org.openhab.core.types.TimeSeries#getPolicy()
+          #   @return [TimeSeries::Policy]
+
+          # @!attribute [r] begin
+          #   Returns the timestamp of the first element in this series.
+          #   @return [Instant]
+
+          # @!attribute [r] end
+          #   Returns the timestamp of the last element in this series.
+          #   @return [Instant]
+
+          # @!attribute [r] size
+          #   Returns the number of elements in this series.
+          #   @return [Integer]
+
+          def initialize(policy)
+            policy = Policy.value_of(policy.to_s.upcase) if policy.is_a?(Symbol)
+            super
+          end
+
+          # Returns true if the series' policy is `ADD`.
+          # @return [true,false]
+          def add?
+            policy == Policy::ADD
+          end
+
+          # Returns true if the series' policy is `REPLACE`.
+          # @return [true,false]
+          def replace?
+            policy == Policy::REPLACE
+          end
+
+          # @!visibility private
+          def inspect
+            "#<OpenHAB::Core::Types::TimeSeries " \
+              "policy=#{policy} " \
+              "begin=#{self.begin} " \
+              "end=#{self.end} " \
+              "size=#{size} " \
+              "states=#{states.inspect}>"
```

Comment:
This seems pretty verbose to include in the inspect output, since it's not just a state, but a full Entry which includes its timestamp

Reply:
I'll remove the states from the inspect. For a "timeseries", the begin/end time is probably more important than the states.